kubemarine/procedures/check_iaas.py

```diff
@@ -258,6 +258,12 @@ def system_distributive(cluster):
             else:
                 detected_supported_os.append(detected_os)
                 cluster.log.debug('Host %s running \"%s\"' % (address, detected_os))
+                result = all(versions == detected_os[0] for versions in detected_os)
```

Comment:
`result` is always false. Even for cluster consisted of the only node. Also, please firstly check that all OS are supported and  raise failure otherwise, and only if all OS are supported, check that they are equal.